Tasks/KubernetesManifestV0/src/utils/CanaryDeploymentHelper.ts

```diff
@@ -207,17 +201,29 @@ function createCanaryObjectsArgumentString(files: string[], includeServices: boo
                 || (includeServices && helper.isServiceEntity(kind))) {
                 const canaryObjectName = getCanaryResourceName(name);
                 const baselineObjectName = getBaselineResourceName(name);
-                kindList.add(kind);
-                nameList.add(canaryObjectName);
-                nameList.add(baselineObjectName);
+                addValueToList(kindNameMap, kind, canaryObjectName);
+                const result = kubectl.getResource(kind, baselineObjectName);
+                if(result != null && !result.stderr)
+                {
+                    addValueToList(kindNameMap, kind, baselineObjectName);
+                }
```

Comment:
Why don't you run the Kubectl delete logic here itself? You don't need to maintain a map then

Reply:
This is done to reduce the number of API calls.